tests/integration/shell/auth.py

```diff
@@ -7,18 +7,37 @@
 # Import python libs
 import os
 import pwd
+import grp
 import random
 
 # Import Salt Testing libs
+from salttesting import skipIf
 from salttesting.helpers import (
     ensure_in_syspath,
     destructiveTest)
 ensure_in_syspath('../../')
 
 # Import salt libs
+from salt.utils.pycrypto import gen_hash
 import integration
 
-from salttesting import skipIf
+
+def gen_password():
+    '''
```

Comment:
Sorry for a nitpick: doc strings should be with triple double-quotes, not single-quotes.